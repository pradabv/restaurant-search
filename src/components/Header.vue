<template>
  <header>
        <div class="container header">
          <h1>{{appHeaderText}}</h1>
        </div>
  </header>
</template>

<script lang="ts">
import { Component,  Vue, Prop } from 'vue-property-decorator';

@Component
export default class Header extends Vue {
   @Prop() private appHeaderText!: string;
}
</script>

<style scoped>
header{
    background-color: #282c34;
    color:#fff;
    padding: 30px 0px;
}
header h1 {
    font-size: 30px;
    padding-left: 30px;
}
</style>


