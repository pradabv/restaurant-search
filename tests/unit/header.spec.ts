import { shallowMount, mount } from '@vue/test-utils';
import Header from '@/components/Header.vue';

describe('Header.vue', () => {
    it('Header renders props.appHeaderText when passed', () => {
        const appHeaderText = 'New Title';
        const wrapper = shallowMount(Header, {
          propsData: { appHeaderText }
        });
        expect(wrapper.text()).toContain(appHeaderText);
    });

    it('Header MatchSnapshot', () => {
      const appHeaderText = 'Restaurant Finder';
      const wrapper = mount(Header, {
        propsData: { appHeaderText }
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
  