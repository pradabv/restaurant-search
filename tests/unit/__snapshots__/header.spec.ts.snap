// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue Header MatchSnapshot 1`] = `
<header>
  <div class="container header">
    <h1>Restaurant Finder</h1>
  </div>
</header>
`;
